igvm/vm.py

```diff
@@ -849,3 +866,83 @@ def vm_performance_value(self) -> float:
         ) * hv_perffactor_src
 
         return float(vm_performance_value)
+
+    def aws_get_instances_overview(self, timeout: int = 5) -> List:
+        """AWS Get Instances Overview
+
+        Load or download the latest instances.json, which contains
+        a complete overview about all instance_types, their configuration,
+        performance and pricing.
+
+        :param: timeout: Timeout value for the get request
+
+        :return: VM types overview as list
+                 or None, if load/download failed
+        """
+
+        url = AWS_INSTANCES_OVERVIEW_URL
+        file = Path(AWS_INSTANCES_OVERVIEW_FILE)
+
+        if file.exists():
+            f = open(AWS_INSTANCES_OVERVIEW_FILE)
+            overview = json.load(f)
+            f.close()
+        else:
+            r = requests.get(url, timeout=timeout)
+            if r.status_code == requests.codes.ok:
+                overview = r.json()
+            else:
+                overview = None
+
+        return overview
+
+    def aws_get_fitting_vm_types(self, overview) -> List:
+        """AWS Get Fitting VM types
+
+        Use the performance_value of the VM and multiply it with a static
+        factor to get the targeted ECU of AWS +-25% (which is kind of a
+        performance_value). Crawl the instances overview for fitting
+        vm_types for this targeted ECU range.
+
+        :param: overview: instances.json with a complete
+                          instance_types overview
+
+        :return: Fitting VM types as list
+        """
+
+        vm_performance_value = self.vm_performance_value()
+        region = self.dataset_obj['aws_placement'][:-1]
```

Comment:
If that is a possible case: What happens when aws_placment attribute is empty (IndexError) ?

Reply:
It's not possible. The placement gets inherited by the network (project_network), which will be/is already prepared in advance. So if you're moving a VM network wise or create a new one within this network, you'll automatically get the needed placement.